src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala

```diff
@@ -638,41 +650,82 @@ final case class TenantApiServiceImpl(
     }
   } yield (updatedTenant, attributeToModify)
 
-  private def assertAttributeVerificationAllowed(producerId: UUID, consumerId: UUID, attributeId: UUID)(implicit
-    contexts: Seq[(String, String)]
-  ): Future[Unit] =
+  private def assertAttributeVerificationAllowed(producerId: UUID, consumerId: UUID, attributeId: UUID): Future[Unit] =
     assertVerifiedAttributeOperationAllowed(
       producerId,
       consumerId,
       attributeId,
-      Seq(AgreementState.PENDING, AgreementState.ACTIVE, AgreementState.SUSPENDED),
+      Seq(AgreementPersistentModel.Pending, AgreementPersistentModel.Active, AgreementPersistentModel.Suspended),
       AttributeVerificationNotAllowed(consumerId, attributeId)
     )
 
-  private def assertAttributeRevocationAllowed(producerId: UUID, consumerId: UUID, attributeId: UUID)(implicit
-    contexts: Seq[(String, String)]
-  ): Future[Unit] = assertVerifiedAttributeOperationAllowed(
-    producerId,
-    consumerId,
-    attributeId,
-    Seq(AgreementState.PENDING, AgreementState.ACTIVE, AgreementState.SUSPENDED),
-    AttributeRevocationNotAllowed(consumerId, attributeId)
-  )
+  private def assertAttributeRevocationAllowed(producerId: UUID, consumerId: UUID, attributeId: UUID): Future[Unit] =
+    assertVerifiedAttributeOperationAllowed(
+      producerId,
+      consumerId,
+      attributeId,
+      Seq(AgreementPersistentModel.Pending, AgreementPersistentModel.Active, AgreementPersistentModel.Suspended),
+      AttributeRevocationNotAllowed(consumerId, attributeId)
+    )
+
+  private def getAgreements(
+    producerId: UUID,
+    consumerId: UUID,
+    agreementStates: Seq[AgreementPersistentModel.PersistentAgreementState],
+    offset: Int,
+    limit: Int
+  )(implicit ec: ExecutionContext): Future[Seq[AgreementPersistentModel.PersistentAgreement]] = {
+    AgreementReadModelQueries.getAgreements(producerId, consumerId, agreementStates, offset, limit)(readModel)
+  }
```

Comment:
Is this required? we can invoke directly `AgreementReadModelQueries.getAgreements`
```suggestion
```